### ISSUE TITLE
KOPS_BASE_URL isn't url-encoded safely when used

### ISSUE THREAD

**Issue by fred-vogt** [kubernetes/kops#9452]
TL;DR - `KOPS_BASE_URL` doesn't work if the URI requires url-encoding, doesn't matter if supplied url-encoded or not.

At issue is that the default CI versioning adds a URL hostile `+` between the bumped version number and git sha.

* [upup/pkg/fi/cloudup/urls.go](https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/urls.go)
* [util/pkg/vfs/context.go](https://github.com/kubernetes/kops/blob/master/util/pkg/vfs/context.go)

This can be easily worked around when building.
But changing `VERSION := ${KOPS_CI_VERSION}+${GITSHA}` to remove the `+` for something that doesn't require escaping would be a good idea.

Makefile:
```
...
ifndef VERSION
 ...
  ifndef CI
    VERSION=${KOPS_RELEASE_VERSION}
  else
    VERSION := ${KOPS_CI_VERSION}+${GITSHA}
  endif
endif
...
```
This can be overriden, but is a bad default since `kops update` will choke on it later if the `CI=1` default version value is used. 

```
kops version --short
1.18.0-beta.2+6309b96  => (url-encoded '+' becomes %2B) 1.18.0-beta.2%2B6309b96
```
NOTE: most tools handle URI url-encoding poorly (curl cannot do it)

You end up with URLs like `https://s3-us-west-2.amazonaws.com/<bucket>/kops/1.18.0-beta.2+6309b96/linux/amd64/nodeup`
These work in with `wget` or `curl` - so its not an issue with bucket access.

Output:
```
Error: Error running command 
KOPS_BASE_URL='https://s3-us-west-2.amazonaws.com/.../kops/1.18.0-beta.2%2B6309b96/' 
KOPS_STATE_S3_ACL='bucket-owner-full-control'
kops update cluster
--name=1-18b2... \
--state=s3://... \
...
': exit status 1. 
Output: W0626 22:42:22.542590     782 urls.go:91] 
  Using base url from KOPS_BASE_URL env var: 
  "https://s3-us-west-2.amazonaws.com/.../kops/1.18.0-beta.2%2B6309b96/"
I0626 22:42:22.659821     782 context.go:272] hit maximum retries 1 with error file does not exist
...
cannot determine hash for "https://s3-us-west-2.amazonaws.com/.../kops/1.18.0-beta.2+6309b96/linux/amd64/nodeup" 
(have you specified a valid file location?)
```

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
Branch: `release-1.18`

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
Custom build from the release branch

**3. What cloud provider are you using?**
AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
`kops update` is what exhibits the issue when checking hashes.

**5. What happened after the commands executed?**
N/A.

**6. What did you expect to happen?**
cluster running with nodeup/assets from a custom build.

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**
N/A.

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**
N/A.

**9. Anything else do we need to know?**
Setting `NODEUP_URL` directly works.  But then it fails on checking hashes for the asset items.

**Comment by fred-vogt** [comment]
Closing.  The git sha is no longer appended with a `+` in master.
